website/blog/2024-11-19-ai-lab-first-app.md

```diff
@@ -0,0 +1,220 @@
+
+We have also indicated our own system prompt:
+
+```
+prompt = ChatPromptTemplate.from_messages([
+    ("system", """
+        reply in JSON format with an array of objects with 2 fields name and url
+        (and with no more text than the JSON output),
+        with a list of pages in the website https://www.podman-desktop.io related to my query
+    """),
+    MessagesPlaceholder(variable_name="history"),
+    ("user", "{input}")
+])
+```
+
+## Testing my own app locally
+
+To iterate during the development of aour app, we can test our app locally in our host system, while using the model served by Podman AI Lab. For this, we need to start a new model service from the page `Models > Services`, by clicking the `New Model Service`, then choosing the appropriate model (`Mistral-7B-instruct` in our case), and specifying a port number (let's say 56625).
+
+![a running inference server with Mistral model](./img/ai-lab-first-app/08-ai-lab-demo-inference-server.png)
+
+Then, we can run our app, by specifying through the `MODEL_ENDPOINT` environment variable how to access the model service.
+
+![my app running locally](./img/ai-lab-first-app/09-ai-lab-demo-my-app-local.png)
+
+Finally, we can send a request to this app running locally, and listening in the port 5000, and we can check in the screenshot below that the response is, as expected, a list of pages (name and url) in JSON format:
+
+![a request to the micro-service](./img/ai-lab-first-app/10-ai-lab-demo-my-app-http-request.png)
+
+## Creating a recipe
+
+The last step is to add this application to the Podman AI Lab recipe catalog.
+
+Podman AI Lab provides a way for a user to extend the provided catalog with its own recipes. This can be done by adding a file in a specific directory, as described [in this documentation](https://github.com/containers/podman-desktop-extension-ai-lab/tree/main?tab=readme-ov-file#-providing-a-custom-catalog).
+
+```
+{
+  "version": "1.0",
+  "recipes": [
+    {
+      "id": "search-podman-desktop-io",
+      "description" : "Search on Podman-desktop.io website",
+      "name" : "Search Podman-desktop.io",
+      "repository": "https://github.com/feloy/podman-ai-lab-demo",
+      "ref": "main",
+      "icon": "natural-language-processing",
+      "categories": [
+        "natural-language-processing"
+      ],
+      "basedir": "",
+      "readme": "",
+      "recommended": [
+        "hf.TheBloke.mistral-7b-instruct-v0.2.Q4_K_M"
+      ],
+      "backend": "llama-cpp"
+    }
+  ]
+}
+```
+
+By adding thie above content to `$HOME/.local/share/containers/podman-desktop/extensions-storage/redhat.ai-lab/user-catalog.json`, you should now be able to see a new recipe `Search Podman-desktop.io` in the recipe catalog of Podman AI Lab, and run it as any other recipe. And, of course, you can share this file with your colleagues to share with them your latest experiment.
```

Comment:
The file is the whole content of the user catalog so it should replace the file and not be appended